.github/workflows/ci.yml

```diff
@@ -262,19 +262,13 @@ jobs:
         run: echo ${{ matrix.env }} >> $GITHUB_ENV
 
       # Build the tests before running to improve cross compilation speed
-      - name: Run cargo build
-        uses: actions-rs/cargo@v1.0.3
-        with:
-          command: build
-          args: --tests ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
-          use-cross: ${{ matrix.target != 'native' }}
+      - name: Run cargo/cross build
+        run: |
+          ${{ matrix.target != 'native' && 'cross' || 'cargo' }} build --tests ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
 
-      - name: Run cargo test
-        uses: actions-rs/cargo@v1.0.3
-        with:
-          command: test
-          args: ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
-          use-cross: ${{ matrix.target != 'native' }}
+      - name: Run cargo/cross test
+        run: |
+          ${{ matrix.target != 'native' && 'cross' || 'cargo' }} test ${{ matrix.exclude }} ${{ matrix.target != 'native' && format('--target {0}', matrix.target) || '' }} ${{ matrix.args }}
```

Comment:
This is a bit verbose. I could split this into different tasks but then we would be repeating ourselves, which given the long list of args seems worse.

```
- name: build with cargo
  if: ${{ matrix.target == 'native' }}
  run: cargo build

- name: build with cross
  if: ${{ matrix.target != 'native' }}
  run: cross build

```

Reply:
lemme experiment with those options.

Reply:
I tried something similar to the env (bash variable) and ran into issues with windows/powershell having trouble parsing the command `$exec build ...`. I can give that one another try.

Always using `cross` seems to work but its a bit weird because it always tries to run in a docker container:
1) on platforms where cross docker images are available, we use the same docker image as the host system
2) on platforms where a cross docker image is not available, we use `cargo` // this is good

1) seems weird because lets say we are on x86_linux and we invoke `cross build`. We will run inside a docker x86_linux docker container. At the very least this is a behavior change from before. Can we expect full parity between docker and the host system? For a networking library I am a bit hesitant.

Reply:
nah just do what we're doing if that's the case